agent/lib/kontena/workers/log_worker.rb

```diff
@@ -56,22 +58,35 @@ def watch_queue
       end
     end
 
-    # Process items from @queue until nil
+    # Process items from @queue
     def process_queue
-      defer {
-        while data = @queue.pop
-          sleep 1 until @processing
-
-          rpc_client.async.notification('/containers/log', [data])
-          if @queue.size > 100
-            sleep 0.001
-          else
-            sleep 0.05
-          end
+      loop do
+        sleep 1 until processing?
+        buffer = @queue.shift(BATCH_SIZE)
+        if buffer.size > 0
+          rpc_client.notification('/containers/log_batch', [buffer])
```

Comment:
Wait, what happaned to the `rpc_client.async.notification`?

I figure there isn't any significant difference though, apart from some possibility of this now raising a `Celluloid::DeadActorError` or `Celluloid::TaskTerminated` if the RpcClient terminates.